Xamarin.PropertyEditing.Windows/PropertyPresenter.cs

```diff
@@ -64,6 +161,54 @@ private void OnLoaded (object sender, RoutedEventArgs e)
 		private void OnUnloaded (object sender, RoutedEventArgs e)
 		{
 			IsSubProperty = false;
+			DataContext = null;
+		}
+
+		private void OnDataContextChanged (object sender, DependencyPropertyChangedEventArgs e)
+		{
+			if (this.pvm != null) {
+				this.pvm.CreateVariantRequested -= OnCreateVariantRequested;
+			}
+
+			this.pvm = e.NewValue as PropertyViewModel;
+			if (this.pvm != null && this.pvm.RequestCreateVariantCommand.CanExecute (null)) {
+				this.pvm.CreateVariantRequested += OnCreateVariantRequested;
+			}
+		}
+
+		private void OnCreateVariantRequested (object sender, CreateVariantEventArgs e)
+		{
+			var variation = CreateVariantWindow.RequestVariant (this, this.pvm.Property);
+			if (variation != null)
+				e.Variation = Task.FromResult (variation);
+		}
+
+		private void OnRemoveVariantClicked (object sender, RoutedEventArgs e)
+		{
+			var parent = (FrameworkElement) VisualTreeHelper.GetParent (this);
+			while (!(parent != null && parent.DataContext is PanelGroupViewModel)) {
```

Comment:
I think we have a helper in `XamlHelper` that looks up the parent chain for a specific type already.

Reply:
I'll swap, I remembered about the XamlHelpers after I wrote a bunch of this; didn't go back and check.

Reply:
It finds a specific view parent type, this is looking for a specific view model type.